src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -253,6 +253,13 @@ public void testRequiredTokenIsEmptyIntArray() throws Exception {
         }
     }
 
+    @Test
+    public void testBehaviourWithZeroChecks() throws Exception {
+        final TreeWalker treeWalker = new TreeWalker();
+        treeWalker.configure(new DefaultConfiguration("default config"));
+        treeWalker.processFiltered(temporaryFolder.newFile("file.java"), new ArrayList<>());
```

Comment:
@ps-sp 
I have already created one issue https://github.com/checkstyle/checkstyle/issues/4128.
Please, create the issue for TreeWalker.